core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java

```diff
@@ -117,11 +117,16 @@ public static GeneticProfile loadGeneticProfile(File file) throws IOException, D
     	if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.GENESET_SCORE) {
             validateGenesetProfile(geneticProfile, file);
         }
+        
+        if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.TREATMENT) {
+            validateTreatment(geneticProfile, file);
+            geneticProfile.setPivotThreshold( Float.parseFloat( geneticProfile.getOtherMetaDataField("pivot_threshold_value") ) );
```

Comment:
These meta fields should be shared between all molecular profiles of GENERIC_ASSAY or not? I guess they should be optional field for any GENERIC_ASSAY profile. These properties are used by the waterfall plot and are optional for TREATMENT_RESPONSE profiles.

And if shared,  we can here remove the differentiation between TREATMENT_RESPONSE and GENERIC_ASSAY can be removed here.